service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/database/CreateAzureDatabaseStep.java

```diff
@@ -0,0 +1,295 @@
+            .identities()
+            .getByResourceGroup(
+                azureCloudContext.getAzureResourceGroupId(),
+                managedIdentityResource.getManagedIdentityName());
+
+    var bearerToken = new BearerToken(samService.getWsmServiceAccountToken());
+    UUID landingZoneId =
+        landingZoneApiDispatch.getLandingZoneId(
+            bearerToken, workspaceService.getWorkspace(workspaceId));
+    var clusterResource =
+        landingZoneApiDispatch
+            .getSharedKubernetesCluster(bearerToken, landingZoneId)
+            .orElseThrow(() -> new RuntimeException("No shared cluster found"));
+
+    var aksApi =
+        kubernetesClientProvider.createCoreApiClient(
+            containerServiceManager,
+            azureCloudContext.getAzureResourceGroupId(),
+            clusterResource);
+    var podName = getPodName(managedIdentity.name());
+    try {
+      startCreateDatabaseContainer(
+          aksApi,
+          bearerToken,
+          landingZoneId,
+          managedIdentity.name(),
+          managedIdentity.principalId(),
+          podName);
+      waitForCreateDatabaseContainer(aksApi, podName);
+    } catch (Exception e) {
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, e);
+    } finally {
+      deleteCreateDatabaseContainer(aksApi, podName);
+    }
+
+    return StepResult.getStepResultSuccess();
+  }
+
+  private void deleteCreateDatabaseContainer(CoreV1Api aksApi, String podName) {
+    try {
+      aksApi.deleteNamespacedPod(podName, aksNamespace, null, null, null, null, null, null);
+    } catch (ApiException e) {
+      logger.warn("Failed to delete create database pod", e);
+    }
+  }
+
+  private void waitForCreateDatabaseContainer(CoreV1Api aksApi, String podName) throws Exception {
+    var finalPhase =
+        RetryUtils.getWithRetry(
+            this::isPodDone,
+            () -> getPodStatus(aksApi, podName),
+            Duration.ofMinutes(10),
+            Duration.ofSeconds(10),
+            0.0,
+            Duration.ofSeconds(10));
+    if (finalPhase.stream().anyMatch(phase -> phase.equals(POD_FAILED))) {
+      throw new RuntimeException("Create database pod failed");
+    }
+  }
+
+  private Optional<String> getPodStatus(CoreV1Api aksApi, String podName) throws ApiException {
+    return Optional.ofNullable(aksApi.readNamespacedPod(podName, aksNamespace, null).getStatus())
+        .map(V1PodStatus::getPhase);
+  }
+
+  private boolean isPodDone(Optional<String> podPhase) {
+    return podPhase
+        .map(phase -> phase.equals(POD_SUCCEEDED) || phase.equals(POD_FAILED))
+        .orElse(false);
+  }
+
+  private void startCreateDatabaseContainer(
+      CoreV1Api aksApi,
+      BearerToken bearerToken,
+      UUID landingZoneId,
+      String newDbUserName,
+      String newDbUserOid,
+      String podName)
+      throws ApiException {
+    try {
+      // TODO: One day we should create our own image for this, but for now we'll use the published
```

Comment:
Kinda neat this works :) 

I suppose in time we could create a Dockerfile, a `set_up_db.sh` script, and a kube `deployment.yaml` to do less of this programmatically.